lib/ssh_pearl.sh

```diff
@@ -18,49 +18,81 @@
     command -v gzip >/dev/null 2>&1 || { echo >&2 "pearl-ssh requires gzip to be installed locally. Aborting."; return 1; }
 
     [[ -z "$PEARL_HOME" ]] && PEARL_HOME=${HOME}/.config/pearl
 
-    local rcScript="$(_aggregate_scripts "$PEARL_HOME/sshrc" "$PEARL_HOME/sshrc.d" | gzip | base64)"
-    local inputrcScript="$(_aggregate_scripts "$PEARL_HOME/sshinputrc" "$PEARL_HOME/sshinputrc.d" | gzip | base64)"
-    local vimrcScript="$(_aggregate_scripts "$PEARL_HOME/sshvimrc" "$PEARL_HOME/sshvimrc.d" | gzip | base64)"
+    [[ "$is_script" -ne 1 && "$__REMOTE_SSH" -ne 1 ]] && rcScript="$(declare -f "$FUNCNAME")"
+    rcScript+="$(_aggregate_scripts "$PEARL_HOME/bashrc" "$PEARL_HOME/bashrc.d")"
+    rcScript="$(echo "$rcScript" | gzip | base64)"
+    inputrcScript="$(_aggregate_scripts "$PEARL_HOME/inputrc" "$PEARL_HOME/inputrc.d" | gzip | base64)"
+    vimrcScript="$(_aggregate_scripts "$PEARL_HOME/vimrc" "$PEARL_HOME/vimrc.d" | gzip | base64)"
+    if [[ "$is_script" -eq 1 ]]; then
+        self="$(gzip < "$0" | base64)"
+        lib="$(gzip < "$(resolve_real_path "$0")/../lib/$LIBNAME" | base64)"
+    fi
 
     CMD="
-        for tmpDir in /tmp \$HOME; do [[ -w \"\$tmpDir\" ]] && { foundTmpDir=\"\$tmpDir\"; break; } done;
-        [[ -z \"\$foundTmpDir\" ]] && { echo >&2 \"Could not find writable tempdirs on the server. Aborting.\"; exit 1; };
+        export __REMOTE_SSH=1  # states we're over ssh session
+        for tmpDir in /tmp \$HOME; do [[ -w \"\$tmpDir\" ]] && { foundTmpDir=\"\$tmpDir\"; break; } done
+        [[ -z \"\$foundTmpDir\" ]] && { echo >&2 \"couldn't find writable tempdirs on the server. Aborting.\"; exit 1; };
         command -v base64 >/dev/null 2>&1 || { echo >&2 \"pearl-ssh requires base64 to be installed on the server. Aborting.\"; exit 1; };
         command -v gunzip >/dev/null 2>&1 || { echo >&2 \"pearl-ssh requires gunzip to be installed on the server. Aborting.\"; exit 1; };
-        PEARLSSH_HOME=\"\$(mktemp -d pearl-XXXXX -p \"\$foundTmpDir\")\";
-        trap \"rm -rf \"\$PEARLSSH_HOME\"; exit\" EXIT HUP INT QUIT PIPE TERM KILL;
-        echo \"${rcScript}\" | base64 -di | gunzip > \"\${PEARLSSH_HOME}/bashrc\";
-        echo \"${inputrcScript}\" | base64 -di | gunzip > \"\${PEARLSSH_HOME}/inputrc\";
-        echo \"${vimrcScript}\" | base64 -di | gunzip > \"\${PEARLSSH_HOME}/vimrc\";
-        VIMINIT=\"let \\\$MYVIMRC='\${PEARLSSH_HOME}/vimrc' | source \\\$MYVIMRC\" INPUTRC=\"\${PEARLSSH_HOME}/inputrc\" bash --rcfile \"\${PEARLSSH_HOME}/bashrc\" -i ${commands_opt};
+        PEARLSSH_ROOT=\"\$(mktemp -d pearl-XXXXX -p \"\$foundTmpDir\")\" || exit 1
+        PEARLSSH_ETC=\"\$PEARLSSH_ROOT/etc\"
+        mkdir \"\$PEARLSSH_ETC\" || exit 1
+        PEARLSSH_BIN=\"\$PEARLSSH_ROOT/bin\"
+        mkdir \"\$PEARLSSH_BIN\" || exit 1
+        PEARLSSH_LIB=\"\$PEARLSSH_ROOT/lib\"
+        mkdir \"\$PEARLSSH_LIB\" || exit 1
+        export PEARL_HOME=\"\$PEARLSSH_ETC\";
```

Comment:
Why should the script look at `~/.config/pearl` on remote system? The whole purpose of pearl-ssh is to take your **own** configuration with you; can't see the benefit of including potentially foreign configuration, that might conflict with your own.